src/catalyst-toolbox/snapshot-lib/src/registration.rs

```diff
@@ -1,9 +1,59 @@
 use jormungandr_lib::crypto::account::Identifier;
 use jormungandr_lib::interfaces::Value;
-use serde::{de::Error, Deserialize, Serialize};
+use serde::{de::Error as _, ser::Error as _, Deserialize, Serialize};
+use std::ops::{Deref, DerefMut};
 
-pub type MainnetRewardAddress = String;
-pub type MainnetStakeAddress = String;
+const MAINNET_PREFIX: &'static str = "addr";
+const TESTNET_PREFIX: &'static str = "addr_test";
+const STAKE_PREFIX: &'static str = "stake";
+
+#[derive(Deserialize, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct MainnetRewardAddress(pub String);
```

Comment:
**Correction**
 We probably don't need to use generic T:

```rust
enum MyWrapper {
    Mainnet(String),
    Testnet(String),
    Stake(String),
}

impl Deref for MyWrapper {
    type Target = String;

    fn deref(&self) -> &Self::Target {
        match self {
            MyWrapper::Mainnet(inner) => inner,
            MyWrapper::Testnet(inner) => inner,
            MyWrapper::Stake(inner) => inner,
        }
    }
}

impl DerefMut for MyWrapper {
    fn deref_mut(&mut self) -> &mut Self::Target {
        match self {
            MyWrapper::Mainnet(inner) => inner,
            MyWrapper::Testnet(inner) => inner,
            MyWrapper::Stake(inner) => inner,
        }
    }
}
```

Reply:
yea, it looks prettier, updating...

Reply:
Ohh, I have tried, unfortunately it is not possible because we need to know which address type is defined during the deserialisation, which is not possible to achieve with enums when we are deserialising just from the `String` type

Reply:
We talked about it privately and came to the conclusion that it's not worth spending more time on it right now.